rust/tests/behaviour/main.rs

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (C) 2022 Vaticle
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+use cucumber::{gherkin::Step, given, then, when, World};
+
+use typeql_lang::parse_query;
+
+#[derive(Debug, Default, World)]
+pub struct TypeQLWorld;
+
+fn main() {
+    futures::executor::block_on(
+        TypeQLWorld::cucumber().filter_run("../vaticle_typedb_behaviour", |_, _, sc| {
+            !sc.tags.iter().any(|t| t == "ignore" || t == "ignore-typeql")
+        }),
+    );
+}
+
+#[given("reasoning schema")]
+#[given("typeql define")]
+#[then("typeql define")]
+#[when("typeql define")]
+async fn typeql_define(_: &mut TypeQLWorld, step: &Step) {
+    let parsed = parse_query(&step.docstring.as_ref().unwrap()).unwrap();
+    assert_eq!(parsed, parse_query(&parsed.to_string()).unwrap());
+}
+
+#[given("typeql undefine")]
+#[when("typeql undefine")]
+async fn typeql_undefine(_: &mut TypeQLWorld, step: &Step) {
+    let parsed = parse_query(&step.docstring.as_ref().unwrap()).unwrap();
+    assert_eq!(parsed, parse_query(&parsed.to_string()).unwrap());
+}
+
+#[given("reasoning data")]
+#[given("typeql insert")]
+#[then("typeql insert")]
+#[when("get answers of typeql insert")]
+#[when("typeql insert")]
+async fn typeql_insert(_: &mut TypeQLWorld, step: &Step) {
+    let parsed = parse_query(&step.docstring.as_ref().unwrap()).unwrap();
+    assert_eq!(parsed, parse_query(&parsed.to_string()).unwrap());
+}
+
+#[given("typeql delete")]
+#[then("typeql delete")]
+#[when("typeql delete")]
+async fn typeql_delete(_: &mut TypeQLWorld, step: &Step) {
+    let parsed = parse_query(&step.docstring.as_ref().unwrap()).unwrap();
+    assert_eq!(parsed, parse_query(&parsed.to_string()).unwrap());
+}
+
+#[when("typeql update")]
```

Comment:
Are we matching the usage of `given`, `when` and `then` to the keywords that they are actually used in in the `behaviour` repo?

I'm inclined to make us not have to worry about it:
```rs
macro_rules! step {
    $str:literal => {
        #[given($str)]
        #[when($str)]
        #[then($str)]
    }
}
```
... assuming that's actually a valid macro by itself.

If it's not possible to make a macro cleanly then we can just leave it as-is. There's just a risk of introducing errors that only get caught at CI time.